test/VetoGuard-Gnosis.test.ts

```diff
@@ -128,12 +130,13 @@ describe("Gnosis Safe", () => {
     vetoERC20Voting = await new VetoERC20Voting__factory(deployer).deploy();
 
     // Deploy VetoGuard contract with a 10 block delay between queuing and execution
-    vetoGuard = await new VetoGuard__factory(deployer).deploy(
-      vetoGuardOwner.address,
-      10,
-      vetoERC20Voting.address,
-      gnosisSafe.address
+    const abiCoder = new ethers.utils.AbiCoder(); // encode data
+    const setupData = abiCoder.encode(
+      ["uint256", "address", "address", "address"],
+      [10, vetoGuardOwner.address, vetoERC20Voting.address, gnosisSafe.address]
     );
+    vetoGuard = await new VetoGuard__factory(deployer).deploy();
+    await vetoGuard.setUp(setupData);
 
     // Initialize VetoERC20Voting contract
     await vetoERC20Voting.initialize(
```

Comment:
It seems like in this test case we're still not getting everything set up in a single transaction yet, right?